### PR TITLE
Fix remove content bug

### LINKED ISSUE #938: Remove table block error
### Description

write a table and a paragraph bellow, select All, press backspace, error happened.

- [x] Can you reproduce the issue? 

### Steps to reproduce

1. create a table. and input some text in every cells.
2. create a paragraph bellow.
3. select the text at the offset 0 in the first cell to the paragraph.
4. press backspace.

**Expected behavior:**

Delete the table and some text in paragraph.

**Actual behavior:**

Error happened, and no longer writable.

**Link to an example: [optional]**

屏幕快照 2019-04-14 上午1 59 17

### Versions

- Mark Text: v0.14.0
- Operating system: All

### PR DESCRIPTION
**Pull request by Jocs**
| Q                | A
| ---------------- | ---
| Bug fix?         | yes
| Fixed tickets    | #934 #938 
| License          | MIT

### Description

1. rewrite selectAll in menu.
   * if the cursor is in a cell, select all only select the full cell.
   * if the cursor is in code block, only select the code block content.
   * if the cursor is in language input, only select the language.

2. valid cursor, if the getCursorRange has invalid node, reset it to a valid cursor.

- [x] FIX #938 

[Description of the bug or feature]

--

### CODE REVIEW

**Comment by Jocs** [comment]
I found another serious bugs, #938 , it's related to this PR, but I think it's not caused by this PR.

File: src/renderer/components/editorWithTabs/editor.vue [hunk 1]
```diff
@@ -557,6 +564,7 @@
       bus.$off('file-loaded', this.setMarkdownToEditor)
       bus.$off('undo', this.handleUndo)
       bus.$off('redo', this.handleRedo)
+      bus.$on('selectAll', this.handleSelectAll)
```
Comment (fxha) [inline src/renderer/components/editorWithTabs/editor.vue]: `bus.$off`

File: src/muya/lib/contentState/backspaceCtrl.js [hunk 2]
```diff
@@ -148,28 +151,38 @@ const backspaceCtrl = ContentState => {
...
+          start: { key, offset },
+          end: { key, offset }
+        }
+        return this.render()
       }
-      return this.render()
     }
 
+
+
```
Comment (fxha) [inline src/muya/lib/contentState/backspaceCtrl.js]: Multiple new lines?

File: src/muya/lib/index.js [hunk 3]
```diff
@@ -45,6 +45,43 @@ class Muya {
...
+    // Select the node that will be observed for mutations
+    const { container } = this
+
+    // Options for the observer (which mutations to observe)
+    const config = { childList: true, subtree: true }
+
+    // Callback function to execute when mutations are observed
+    const callback = function(mutationsList, observer) {
+      for(const mutation of mutationsList) {
+        if (mutation.type == 'childList') {
```
Comment (fxha) [inline src/muya/lib/index.js]: Did you mean `===`?

File: src/muya/lib/index.js [hunk 4]
```diff
@@ -45,6 +45,43 @@ class Muya {
...
+          if (removedNodes && removedNodes.length) {
+            const hasTable = Array.from(removedNodes).some(node => node.nodeType === 1 && node.closest('table.ag-paragraph'))
+            if (hasTable) {
+              console.warn('There was a problem with the table deletion.')
+            }
+          }
+
+          if (target.getAttribute('id') === 'ag-editor-id' && target.childElementCount === 0) {
+            // TODO: the editor can not be input any more. report bugs and recoveryr...
+            console.warn('editor crashed, and can not be input any more.')
```
Comment (fxha) [inline src/muya/lib/index.js]: Can we add more information to better reproduce this?

Comment (Jocs) [reply src/muya/lib/index.js]: It seems that there is no more information now, because we have not recorded the user's operation, etc.

**Pull request merged**